src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html

```diff
@@ -93,6 +93,35 @@
           </div>
         </div>
 
+        <!-- Placement target -->
+        <div class="form-group"
+             [ngClass]="{'has-error': bucketForm.showError('placement-target', frm)}"
```

Comment:
Right now, changing to: ChangeDetectionStrategy.OnPush

has side effects regarding the bucket editing part, so additional refactor is needed.

@tspmelo should I address this for this form on this PR?

Reply:
Since this is a global problem I would suggest doing it in another PR.
Codelyzer already has a rule for that, we just need to activate it [[1]](http://codelyzer.com/rules/prefer-on-push-component-change-detection/).

Regarding methods in html, a better solution is to use pipes.
They handle changes better and are only triggered when the value changes.
Its not always possible to replace methods with a pipe, but we should do it when possible.

Regarding `showError` I have removed it from our codebase in the bootstrap PR, we now use a service that handles the validation automatically.
But I guess we should check if the problem still happens there.